generators/entity-client/templates/react/src/main/webapp/app/entities/entity-dialog.tsx.ejs

```diff
@@ -139,7 +139,15 @@ export class <%= entityReactName %>Dialog extends React.Component<I<%= entityRea
     this.props.<%= val.action %>();
     <%_ }) _%>
   }
-
+  <%_ if (hasRelationshipQuery) { _%>
+  clearEmptyRelations = entity => {
+    Object.keys(entity).filter(k => entity[k] instanceof Object).forEach(k => {
+      if (entity[k]['id'] === '') {
+        delete entity[k];
```

Comment:
React folks will kill us if they see use of delete, so no, lets not do this. Mutating stuff in React is heavily frowned upon

Reply:
could you explain the issue to me, maybe there is another way

Reply:
When you don't put any value in a relationship field in a form, it creates a field in values that looks like this: `values.<relationship>.id: ""`. This by itself is no big deal but once you send that to the server it returns a 500 thinking that the id is invalid. I could try to create a new clean object for the server in the reducer right before calling axios. wdyt ?

Reply:
Can we check why it creates the empty id id field in the first place? May be we can avoidnit from happening unless a relationship is selected

Reply:
I think it's caused by AvForm.